pkg/stanza/fileconsumer/matcher/matcher.go

```diff
@@ -63,7 +73,9 @@ func New(c Criteria) (*Matcher, error) {
 		}, nil
 	}
 
-	if c.OrderingCriteria.Regex == "" {
+	regexIsUsed := orderingCriteriaNeedsRegex(c.OrderingCriteria.SortBy)
+
+	if regexIsUsed && c.OrderingCriteria.Regex == "" {
```

Comment:
I think we could consolidate this bit of logic with the part below which compiles the regex. The positioning of the checks in between appear to be arbiratry.

Reply:
Good point, I've condensed these blocks